### ISSUE TITLE
`no-implicit-reexport` ignores issues after calling `numpy.cross` in a function

### ISSUE THREAD

**Issue by ThunderKey** [python/mypy#12350]
**Bug Report**

When passing the `--no-implicit-reexport` flag to mypy the following code does not generate any errors and does not reveal a type:

```python
import numpy
from numpy.typing import ArrayLike


def test_method() -> None:
    value: ArrayLike = numpy.cross(numpy.array([1, 2, 3]), numpy.array([4, 5, 6]))
    reveal_type(value)
    broken: int = 'test'
```

The issue remains if the method is called directly as `numpy.core.numeric.cross`.

If the flag `--no-implicit-reexport` is not supplied the errors are generated correctly.

Interestingly if the code is not in a method the error is reported. This made me think that the issue might not be with `numpy` but `mypy`.

**To Reproduce**

1. Install dependencies `pip install mypy==0.940 numpy==1.22.3`
1. Create a file `test.py` as mentioned above
1. `mypy test.py` -> errors are shown
1. `mypy --no-implicit-reexport test.py` -> no errors are shown

**Expected Behavior**

Both calls should raise the same errors.

**Actual Behavior**

The call with the `--no-implicit-reexport` flag does not show any errors.

**Your Environment**

- Mypy version used: 0.940 and 0.931
- Mypy configuration options from `mypy.ini` (and other config files): No configuration file
- Python version used: 3.10.2
- Operating system and version: `Linux-5.10.102-1-MANJARO-x86_64-with-glibc2.35`(as returned by `platform.platform()`) but it also fails with the docker image `python:3.10.2-slim`

**Comment by hauntsaninja** [comment]
Interesting. mypy seems to think that code is unreachable, which is why you don't get errors for it. Using `--warn-unreachable` will confirm this.

This seems to be because mypy infers `numpy.cross(numpy.array([1, 2, 3]), numpy.array([4, 5, 6])` as returning the bottom type.

**Comment by hauntsaninja** [comment]
My guess is it's a mistake in numpy's stubs: https://github.com/numpy/numpy/blob/71e7620711d6ff8cb838bdffb5e91cdd25497dba/numpy/core/numeric.pyi#L469
Probably something along the lines of `_ArrayLikeBool_co`'s definition relying on an implicit export, which gets resolved to Any, which causes every overload to match^

**Comment by ThunderKey** [comment]
Thank you @hauntsaninja, that's a very good catch!

I will create an issue in the numpy repo and from now on enable `--warn-unreachable` to detect similar issues.

I still find it a bit strange, that if the code is not in a function the issue does not exist, but for me the workaround works.
